xmtp_api_http/src/http_client.rs

```diff
@@ -0,0 +1,59 @@
+use crate::{HttpClientError, XmtpHttpApiClient};
+use bytes::Bytes;
+use std::pin::Pin;
+use xmtp_proto::traits::{ApiError, Client};
+
+impl From<HttpClientError> for ApiError<HttpClientError> {
+    fn from(value: HttpClientError) -> Self {
+        ApiError::Client { source: value }
+    }
+}
+
+#[cfg_attr(not(target_arch = "wasm32"), async_trait::async_trait)]
+#[cfg_attr(target_arch = "wasm32", async_trait::async_trait(?Send))]
+impl Client for XmtpHttpApiClient {
+    type Error = HttpClientError;
+    type Stream = Pin<Box<dyn futures::Stream<Item = Result<Bytes, HttpClientError>> + Send>>;
+    async fn request(
+        &self,
+        request: http::request::Builder,
+        body: Vec<u8>,
+    ) -> Result<http::Response<Bytes>, ApiError<Self::Error>> {
+        let request = request.body(body.clone())?;
+        let (parts, _) = request.into_parts();
+
+        let req = self
+            .http_client
+            .request(parts.method.clone(), parts.uri.to_string());
```

Comment:
the method for grpc requests will always be `POST`. I think we can hard code that and not worry about it in the `Query` trait implementation